source/mqtt_client_connection.c

```diff
@@ -643,6 +644,10 @@ static void s_suback_callback(
     (void)connection;
 
     PyObject *callback = userdata;
+    if (!callback) {
+        return;
```

Comment:
The way your code is now, you want to either:
- check `if(callback != Py_None)`
- OR when you're passing the userdata to C, do `(callback == Py_None) ? NULL : callback`

I personally prefer the 1st approach, but it's apples and oranges